connection.go

```diff
@@ -33,13 +33,28 @@ func (c *conn) ExecContext(ctx context.Context, query string, args []driver.Name
 		panic("database/sql/driver: misuse of duckdb driver: ExecContext after Close")
 	}
 
-	if len(args) == 0 {
-		// This should be removed once duckdb_extract_statements and related APIs are available to parse multiple statements
-		// so query cancellation works for this case as well
-		return c.execUnprepared(query)
+	stmts, size, err := c.extractStmts(query)
+	if err != nil {
+		return nil, err
+	}
+	defer C.duckdb_destroy_extracted(&stmts)
+
+	// execute all statements without args, except the last one
```

Comment:
What is your idea behind preparing the last instead of the first statement with the given args? Or is this a common behaviour around `database/sql` drivers?

Reply:
This behaviour is same as [Python](https://github.com/duckdb/duckdb/blob/master/tools/pythonpkg/src/pyconnection.cpp#L440) and [C](https://github.com/duckdb/duckdb/blob/master/test/api/capi/test_capi_extract.cpp#L6) driver.

Reply:
Thanks for the explanation. What do you think would be the best approach?

1. Distribute all arguments over all statements so that we can do sth. like `INSERT INTO foo VALUES(?); INSERT INTO bar VALUES(?)` with two args (1, 2).
2. Only apply the args to the last statement (like it is ATM).

My preference would be 1. to be as flexible as possible and to have a predictable behaviour. Otherwise we have to document, that the given args are only applied to the last statement.

Reply:
Okay! I think we then should document this behaviour, to make it clear, that args are only applied to the last statement.